android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiNestedListView.java

```diff
@@ -0,0 +1,448 @@
+	 */
+	public boolean isTouchScrollable()
+	{
+		return this.isTouchScrollable;
+	}
+
+	/**
+	 * Sets whether or not the end-user can scroll the list view with touches.
+	 * <p>
+	 * Note that the list view can still be scrolled programmatically if this setting is false.
+	 * @param value Set true (the default) to allow the user to scroll with touches. Set false to disable.
+	 */
+	public void setTouchScrollable(boolean value)
+	{
+		this.isTouchScrollable = value;
+	}
+
+	/**
+	 * Called before onTouchEvent() gets called in this view or a child view,
+	 * but only while requestDisallowInterceptTouchEvent() is set to false.
+	 * <p>
+	 * Provides this view the opportunity to monitor or steal child touch events.
+	 * If this view returns true, then the event will not be received by the child.
+	 * @param event The touch information received.
+	 * @return Returns true if this view will claim the event. Returns false if not.
+	 */
+	@Override
+	public boolean onInterceptTouchEvent(MotionEvent event)
+	{
+		// If touch scrolling is disabled, then block touch move/drag events.
+		if (!this.isTouchScrollable && (event.getActionMasked() == MotionEvent.ACTION_MOVE)) {
+			return false;
+		}
+
+		// Let the base class handle the touch event.
+		return super.onInterceptTouchEvent(event);
+	}
+
+	/**
+	 * Called when a touch event has been received by this view.
+	 * @param event The touch information received.
+	 * @return Returns true if this view handled the touch. Returns false if not.
+	 */
+	@Override
+	public boolean onTouchEvent(MotionEvent event)
+	{
+		// If touch scrolling is disabled, then block touch move/drag events.
+		if (!this.isTouchScrollable && (event.getActionMasked() == MotionEvent.ACTION_MOVE)) {
+			return false;
+		}
+
+		// Let the base class handle the touch event.
+		return super.onTouchEvent(event);
+	}
+
+
+	/** Creates a new list view. */
+	public static TiNestedListView createUsing(Context context)
+	{
+		if (Build.VERSION.SDK_INT >= 21) {
+			return new ApiLevel21.NestedListView(context);
+		}
+		return new ApiLevel16.NestedListView(context);
+	}
+
+	/** Creates a new list view. */
+	public static TiNestedListView createUsing(Context context, AttributeSet attributeSet)
+	{
+		if (Build.VERSION.SDK_INT >= 21) {
+			return new ApiLevel21.NestedListView(context, attributeSet);
+		}
+		return new ApiLevel16.NestedListView(context, attributeSet);
+	}
+
+	/** Creates a new list view. */
+	public static TiNestedListView createUsing(
+		Context context, AttributeSet attributeSet, int defaultStyleAttributes)
+	{
+		if (Build.VERSION.SDK_INT >= 21) {
+			return new ApiLevel21.NestedListView(context, attributeSet, defaultStyleAttributes);
+		}
+		return new ApiLevel16.NestedListView(context, attributeSet, defaultStyleAttributes);
+	}
+
+
+	/** Provides access to API Level 16 features. */
+	private static class ApiLevel16
+	{
+		/** Constructor made private to prevent instances from being made. */
+		private ApiLevel16() {}
```

Comment:
Minor, you can make this an abstract class to prevent instances from being made instead of creating a private constructor.
```Java
private static abstract class ApiLevel16
```